mech_client/interact.py

```diff
@@ -281,4 +267,9 @@ def interact(
         prompt=prompt,
         tool=tool,
     )
-    return watch_for_events(wss=wss, mech_contract=mech_contract, ledger_api=ledger_api)
+    request_id = watch_for_request_id(
+        wss=wss, mech_contract=mech_contract, ledger_api=ledger_api
+    )
+    print(f"Created on-chain request with ID {request_id}")
+    data = wait_for_data(crypto=crypto)
```

Comment:
Imo we should wait for data here from ACN and on-chain concurrently and then print whatever arrives first and then what arrives second